x/bank/types/balance.go

```diff
@@ -64,9 +64,14 @@ func SanitizeGenesisBalances(balances []Balance) []Balance {
 
 	// 1. Retrieve the address equivalents for each Balance's address.
 	addresses := make([]sdk.AccAddress, len(balances))
+	existingAddresses := map[string]bool{}
```

Comment:
```go
// Track any duplicate addresses for security checks to
// avoid false positives on invariant checks.
seen := make(map[string]bool, len(addresses))
```

Reply:
To add to this, genesis can be very large. To avoid allocating booleans, just use `struct{}{}`.

Reply:
bump :)